mystring.h

```diff
@@ -1,7 +1,19 @@
 #include <iostream>
 #include <memory>
 
-class String {
+class StringInterface {  
```

Comment:
also it's more typical to name interfaces in some clear (but not overwhelming) pattern, here I suggest simply "IString"

Reply:
done